tests/ops/op_math/test_controlled.py

```diff
@@ -2058,38 +2058,6 @@ def stopping_condition(o):
         expected_mat = qml.matrix(expected, wire_order=[3, 7, 0])
         assert qml.math.allclose(actual_mat, expected_mat, atol=tol, rtol=0)
 
-    @pytest.mark.parametrize(
-        "op",
-        [
-            qml.ctrl(qml.ctrl(qml.S, 7), 3),  # nested control
-            qml.ctrl(qml.S, [3, 7]),  # multi-wire control
-        ],
-    )
-    def test_nested_ctrl_containing_phase_shift(self, op):
-        """Test that nested controlled ops are expanded correctly when phase shift is involved
-
-        The decomposition of S gate contains a PhaseShift. In the nested case, we do not want
-        to apply control to the expanded PhaseShift like how it is typically done for other
-        operations, because the decomposition of PhaseShift contains a GlobalPhase, the controlled
-        version of which we do not have handling for.
-
-        TODO: remove this special case once ControlledGlobalPhase is implemented.
```

Comment:
The premise of this entire test was that since `Controlled(GlobalPhase)` couldn't be decomposed, we try to avoid producing it in the first place. This is no longer the case, a `Controlled(GlobalPhase)` is decomposed like any other operator. This test does not need to exist anymore.